signer/src/storage/postgres.rs

```diff
@@ -1701,6 +1702,99 @@ impl super::DbRead for PgStore {
 
         self.get_sweep_transaction_by_txid(&txid).await
     }
+
+    async fn get_latest_unconfirmed_sweep_transactions(
+        &self,
+        chain_tip: &model::BitcoinBlockHash,
+        context_window: u16,
+        prevout_txid: &model::BitcoinTxId,
+    ) -> Result<Vec<model::SweepTransaction>, Error> {
```

Comment:
nit: this fn may return a suffix of the chain of sweep tx starting from `prevout_txid`, if some prefix of the chain is confirmed. The result is that the first sweep tx of the vec may not be spending `prevout_txid`. I guess it's fine with its current usage, but I wanted to point it out in case we plan to rely on that assumption (feel free to resolve this convo).